ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -210,33 +229,70 @@ const Footer = () => {
     onCancel({ location: MetaMetricsEventLocation.Confirmation });
   }, [onCancel, shouldThrottleOrigin]);
 
+  const isShowShieldFooterCoverageIndicator = useEnableShieldCoverageChecks();
+
   return (
-    <PageFooter
-      className="confirm-footer_page-footer"
-      flexDirection={FlexDirection.Column}
-    >
-      <OriginThrottleModal
-        isOpen={showOriginThrottleModal}
-        onConfirmationCancel={onCancel}
-      />
-      <Box display={Display.Flex} flexDirection={FlexDirection.Row} gap={4}>
-        <Button
-          block
-          data-testid="confirm-footer-cancel-button"
-          onClick={handleFooterCancel}
-          size={ButtonSize.Lg}
-          variant={ButtonVariant.Secondary}
-        >
-          {t('cancel')}
-        </Button>
-        <ConfirmButton
-          alertOwnerId={currentConfirmation?.id}
-          onSubmit={() => onSubmit()}
-          disabled={isConfirmDisabled}
-          onCancel={onCancel}
+    <>
+      {isShowShieldFooterCoverageIndicator && <ShieldFooterCoverageIndicator />}
+      <PageFooter
+        className="confirm-footer_page-footer"
+        flexDirection={FlexDirection.Column}
+        // box shadow to match the original var(--shadow-size-md) on the footer,
+        // but only applied to the bottom of the box, so it doesn't overlap with
+        // the shield footer coverage indicator
+        style={
+          isShowShieldFooterCoverageIndicator
+            ? { boxShadow: '0 4px 16px -8px var(--color-shadow-default)' }
+            : undefined
+        }
+      >
+        <OriginThrottleModal
+          isOpen={showOriginThrottleModal}
+          onConfirmationCancel={onCancel}
         />
-      </Box>
-    </PageFooter>
+        <Box display={Display.Flex} flexDirection={FlexDirection.Row} gap={4}>
+          {currentConfirmation?.type !==
+            TransactionType.shieldSubscriptionApprove && (
+            <Button
+              block
+              data-testid="confirm-footer-cancel-button"
+              onClick={handleFooterCancel}
+              size={ButtonSize.Lg}
+              variant={ButtonVariant.Secondary}
+            >
+              {t('cancel')}
+            </Button>
+          )}
+          <ConfirmButton
+            alertOwnerId={currentConfirmation?.id}
+            onSubmit={onSubmit}
+            disabled={isConfirmDisabled}
+            onCancel={onCancel}
+          />
+        </Box>
+        {currentConfirmation?.type ===
+          TransactionType.shieldSubscriptionApprove && (
+          <Box display={Display.Flex} flexDirection={FlexDirection.Row} gap={4}>
```

Comment:
`ShieldAgreeButton` component in another file to decouple shield from generic footer?